hawkular-wildfly-agent-installer/src/main/java/org/hawkular/wildfly/agent/installer/TargetConfigInfo.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.wildfly.agent.installer;
+
+public interface TargetConfigInfo {
+    /**
+     * The root that all other xpaths are relative to.
+     */
+    String getRootXPath();
+
+    /**
+     * @return the XPath that refers to the security-realms element in the config file.
+     */
+    String getSecurityRealmsXPath();
+
+    /**
+     * @return the XPath that refers to the profile element in the config file.
+     */
+    String getProfileXPath();
+
+}
```

Comment:
if you were interested in using java8 default methods this might be a decent place to do it since the overriding impls are often the same.